android/src/main/java/com/reactnativestripesdk/pushprovisioning/TapAndPayProxy.kt

```diff
@@ -0,0 +1,133 @@
+    }
+  }
+
+  fun tokenize(activity: Activity, tokenReferenceId: String, token: ReadableMap, cardDescription: String) {
+    try {
+      val tapAndPayClientClass = Class.forName("com.google.android.gms.tapandpay.TapAndPayClient")
+      val tokenizeMethod = tapAndPayClientClass::class.java.getMethod("tokenize", Activity::class.java, String::class.java, Int::class.java, String::class.java, Int::class.java, Int::class.java)
+      tokenizeMethod.invoke(tapAndPayClient,
+                            activity,
+                            tokenReferenceId,
+                            token.getInt("serviceProvider"),
+                            cardDescription,
+                            token.getInt("network"),
+                            REQUEST_CODE_TOKENIZE)
+    } catch (e: Exception) {
+      Log.e(TAG, "Google TapAndPay dependency not found.")
+    }
+  }
+
+  private fun createResult(cardIsInWallet: Boolean, token: Any? = null): WritableMap {
+    val result = WritableNativeMap()
+    result.putBoolean("isInWallet", cardIsInWallet)
+    result.putMap("token", mapFromTokenInfo(token))
+    return result
+  }
+
+  private fun mapFromTokenInfo(token: Any?): WritableMap? {
+    if (token == null) {
+      return null
+    }
+    val result = WritableNativeMap()
+    try {
+      val tokenInfoClass = Class.forName("com.google.android.gms.tapandpay.issuer.TokenInfo")
+      result.putString(
+        "id",
+        tokenInfoClass.getMethod("getIssuerTokenId").invoke(token) as String)
+      result.putString(
+        "cardLastFour",
+        tokenInfoClass.getMethod("getFpanLastFour").invoke(token) as String)
+      result.putString(
+        "issuer",
+        tokenInfoClass.getMethod("getIssuerName").invoke(token) as String)
+      result.putString(
+        "status",
+        mapFromTokenState(tokenInfoClass.getMethod("getTokenState").invoke(token) as Int))
+      result.putInt(
+        "network",
+        tokenInfoClass.getMethod("getNetwork").invoke(token) as Int)
+      result.putInt(
+        "serviceProvider",
+        tokenInfoClass.getMethod("getTokenServiceProvider").invoke(token) as Int)
+    } catch (e: Exception) {
+      Log.e(TAG,
+            "There was a problem finding the class com.google.android.gms.tapandpay.issuer.TokenInfo. Make sure you've included Google's TapAndPay dependency.")
+    }
+    return result
+  }
+
+  private fun mapFromTokenState(status: Int): String {
+    return when (status) {
+      getTokenState("TOKEN_STATE_NEEDS_IDENTITY_VERIFICATION") -> "TOKEN_STATE_NEEDS_IDENTITY_VERIFICATION"
+      getTokenState("TOKEN_STATE_PENDING") -> "TOKEN_STATE_PENDING"
+      getTokenState("TOKEN_STATE_SUSPENDED") -> "TOKEN_STATE_SUSPENDED"
+      getTokenState("TOKEN_STATE_ACTIVE") -> "TOKEN_STATE_ACTIVE"
+      getTokenState("TOKEN_STATE_FELICA_PENDING_PROVISIONING") -> "TOKEN_STATE_FELICA_PENDING_PROVISIONING"
+      getTokenState("TOKEN_STATE_UNTOKENIZED") -> "TOKEN_STATE_UNTOKENIZED"
```

Comment:
Great catch, since these are ints we actually need to use reflection again 🪞 

fixed here-
https://github.com/stripe/stripe-react-native/pull/889/commits/3259f1d15a2a2c1594951405d0caab8c8c1e2c59